src/proc/neon/image-neon.cpp

```diff
@@ -0,0 +1,215 @@
+                        r16.val[j] = vmaxq_s16(n0, r16.val[j]);
+
+                        g16.val[j] = vsubq_s16(
+                            vsubq_s16(c, vshrq_n_s16(vmulq_s16(d, vdupq_n_s16(100)), 8)),
+                            vshrq_n_s16(vmulq_s16(e, vdupq_n_s16(208)), 8)
+                        );
+                        g16.val[j] = vmaxq_s16(n0, g16.val[j]);
+
+                        b16.val[j] = vaddq_s16(
+                            c,
+                            vaddq_s16(
+                                vshrq_n_s16(vmulq_s16(d, vdupq_n_s16(516 - (1 << 8))), 8),
+                                d
+                            )
+                        );
+                        b16.val[j] = vmaxq_s16(n0, b16.val[j]);
+                    }
+
+                    // int16 -> uint8 and combine x4x4 to x16
+                    r8 = vcombine_u8(
+                        vqmovn_u16((uint16x8_t)r16.val[0]),
+                        vqmovn_u16((uint16x8_t)r16.val[1])
+                    );
+                    g8 = vcombine_u8(
+                        vqmovn_u16((uint16x8_t)g16.val[0]),
+                        vqmovn_u16((uint16x8_t)g16.val[1])
+                    );
+                    b8 = vcombine_u8(
+                        vqmovn_u16((uint16x8_t)b16.val[0]),
+                        vqmovn_u16((uint16x8_t)b16.val[1])
+                    );
+                }
+
+                if (FORMAT == RS2_FORMAT_RGBA8)
+                {
+                    uint8x16x4_t rgba;
+                    rgba.val[0] = r8;
+                    rgba.val[1] = g8;
+                    rgba.val[2] = b8;
+                    rgba.val[3] = vdupq_n_u8(255);
+                    vst4q_u8(&dst[i * 4], rgba);
+                    continue;
+                }
+                if (FORMAT == RS2_FORMAT_RGB8)
+                {
+                    uint8x16x3_t rgb;
+                    rgb.val[0] = r8;
+                    rgb.val[1] = g8;
+                    rgb.val[2] = b8;
+                    vst3q_u8(&dst[i * 3], rgb);
+                    continue;
+                }
+                if (FORMAT == RS2_FORMAT_BGRA8)
+                {
+                    uint8x16x4_t bgra;
+                    bgra.val[0] = b8;
+                    bgra.val[1] = g8;
+                    bgra.val[2] = r8;
+                    bgra.val[3] = vdupq_n_u8(255);
+                    vst4q_u8(&dst[i * 4], bgra);
+                    continue;
+                }
+                if (FORMAT == RS2_FORMAT_BGR8)
+                {
+                    uint8x16x3_t bgr;
+                    bgr.val[0] = b8;
+                    bgr.val[1] = g8;
+                    bgr.val[2] = r8;
+                    vst3q_u8(&dst[i * 3], bgr);
+                    continue;
+                }
+            }
+        }
+
+        void unpack_yuy2_neon_y8(uint8_t * const d[], const uint8_t * s, int n)
+        {
+            unpack_yuy2_neon<RS2_FORMAT_Y8>(d, s, n);
```

Comment:
Why not calling the templated function at the first place instead of adding functions that call it?

Reply:
I used the code image-avx.cpp as a reference, but is this wrong?
https://github.com/IntelRealSense/librealsense/blob/e1688cc318457f7dd57abcdbedd3398062db3009/src/image-avx.cpp#L253

Reply:
Not wrong.. just redundant.|
But since our code is a reference let's keep as is